Python/gc.c

```diff
@@ -1384,12 +1385,9 @@ expand_region_transitively_reachable(PyGC_Head *container, PyGC_Head *gc, GCStat
 static void
 completed_cycle(GCState *gcstate)
 {
-    PyGC_Head *not_visited = &gcstate->old[gcstate->visited_space^1].head;
-    assert(gc_list_is_empty(not_visited));
+    assert(gc_list_is_empty(&gcstate->old[gcstate->visited_space^1].head));
```

Comment:
FWIW, `gcstate->old[gcstate->visited_space^1]` is about as unclear to me it could possibly be, as a reader with very little context about the GC implementation.  I'm not saying you need to do anything about this, but just wanted to point out my experience, which I expect most people would share.  (I do suppose that the GC implementation is saturated with tricky code that requires sufficient fore-knowledge to follow.)

Is there any single document that condenses all the context into one spot?  If not, it might be worth adding one, or at least adding comments where necessary to give at least some context for the trickier spots.

Reply:
I changed this to avoid the unused variable error.
I've changed it back and added an `#ifdef` around it.